Release April 2024

MakisH opened the pull request:
This is a big update, which:

- moves to SU2 7.5 and completely revamps the adapter to use the Python wrapper of SU2 (+ small changes in the C++ code, just for checkpointing), instead of directly changing multiple SU2 source files. #32 #34 
- moves to the preCICE v3 API. #33 

This state is compatible with the tutorials PR https://github.com/precice/tutorials/pull/324.

Still needs updates for the documentation on the website (or make the website source them from here).

No versioning system is established at the moment, but this is now pointing towards v1.0 @uekerman (assuming 0.x would have been the previous state). We could still now start with v0.2 to be conservative, or keep using untagged releases. But I think we should take the opportunity to mark this with a clear version.

BIG thanks to @j-signorelli for his excellent contribution and his patience getting that reviewed and merged.

uekerman commented:
This new version now still has a small part that acts as a patch to the SU2 sources, right? Does this mean that we likely need to update it for every new SU2 version, similar to e.g. the CalculiX adapter? Then, maybe, a versioning scheme following the SU2 versions could be helpful?
Otherwise, v1.0 could be a good option.

MakisH commented:
Good point about the SU2 replacement files. That might still change soon.
On the other hand, if we now go with the SU2 versions and then want to switch to another system, we will have a version sorting issue.

Let's just keep this as-is for now (commit id), and decide later.

MakisH commented:
I updated the documentation as well, which is now sourced from this repository (develop branch): https://precice.org/adapter-su2-overview.html You might need to refresh each page.

I have also integrated the recent changed of @BenjaminRodenberg and changed the default branch to `develop`.

This is now ready to merge from my side.

IshaanDesai reviewed (approved):
I looked over the files, but did not test anything, as sufficient testing has already been done.

uekerman commented on run/SU2_preCICE_CHT.py:
```diff
@@ -0,0 +1,283 @@
+## \file launch_unsteady_CHT_FlatPlate.py
+#  \brief Python script to launch SU2_CFD with customized unsteady boundary conditions using the Python wrapper.
+#  \author David Thomas
+#  \version 7.4.0 "Blackbird"
```
Seems to be an older copy?

MakisH replied:
I think this is just the starting file attribution, I did not want to change that without deeply comparing the original files between versions.

@j-signorelli do you think we should change this comment to a newer version, or even just remove it? Do you know if SU2 7.5.0 is supported, or should we mention only 7.5.1 everywhere?

I can figure these all out by myself, but I would prefer a simple affirmation in case you already know.

j-signorelli replied:
@MakisH that's just remnants of the original file I used as a starting point for implementation -- it's probably completely fine to just remove it altogether. It has been quite awhile so I don't remember but I do think SU2 v7.5.0 should also work just fine for this.

MakisH replied:
Updated the headers in https://github.com/precice/su2-adapter/pull/40
I only need an approval in that, and will merge the release PR after that.

uekerman reviewed (approved):
Thanks a lot @MakisH :pray: 

I looked at the docs and the diff. All looks reasonable. I did not run anything.

@MakisH Are you sure that v7.5.0 is still supported? The replacement file is apparently from v7.5.1. Could be that there are simply no changes in that file, but could be safer to simply remove the statement that it also supports v7.5.0.
